frontend/amundsen_application/api/metadata/v0.py

```diff
@@ -842,3 +851,69 @@ def get_column_lineage() -> Response:
     except Exception as e:
         payload = jsonify({'msg': 'Encountered exception: ' + str(e)})
         return make_response(payload, HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@metadata_blueprint.route('/feature', methods=['GET'])
+def get_feature_metadata() -> Response:
+    """
+    call the metadata service endpoint and return matching results
+    :return: a json output containing a feature metadata object as 'featureData'
+
+    """
+    try:
+        feature_key = get_query_param(request.args, 'key')
+        list_item_index = request.args.get('index', None)
+        list_item_source = request.args.get('source', None)
+
+        results_dict = _get_feature_metadata(feature_key=feature_key, index=list_item_index, source=list_item_source)
+        return make_response(jsonify(results_dict), results_dict.get('status_code', HTTPStatus.INTERNAL_SERVER_ERROR))
+    except Exception as e:
+        message = 'Encountered exception: ' + str(e)
+        logging.exception(message)
+        return make_response(jsonify({'featureData': {}, 'msg': message}), HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+@action_logging
+def _get_feature_metadata(*, feature_key: str, index: int, source: str) -> Dict[str, Any]:
+
+    results_dict = {
+        'featureData': {},
+        'msg': '',
+    }
+
+    try:
+        feature_endpoint = _get_feature_endpoint()
+        url = '{0}/{1}'.format(feature_endpoint, feature_key)
+        response = request_metadata(url=url)
+    except ValueError as e:
+        # envoy client BadResponse is a subclass of ValueError
+        message = 'Encountered exception: ' + str(e)
+        results_dict['msg'] = message
+        results_dict['status_code'] = getattr(e, 'code', HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.exception(message)
+        return results_dict
+
+    status_code = response.status_code
+    results_dict['status_code'] = status_code
+
+    if status_code != HTTPStatus.OK:
+        message = 'Encountered error: Metadata request failed'
+        results_dict['msg'] = message
+        logging.error(message)
+        return results_dict
+
+    try:
+        feature_data_raw: dict = response.json()
+
+        feature_data_raw['key'] = feature_key
```

Comment:
Adding the 'key' and the 'marshall' functions in the frontend were added because FE needed some things that the metadata or search service didn't include. I talked about this with @feng-tao a while ago. What do you think about adding these types of fields to the upstream responses?


(Not necessarily asking to change these here, just starting a discussion) 
(Also wondering if FE should just call search/metadata services directly instead of having a FE Flask layer)

Reply:
Honestly I have been thinking the same thing, its all basically just duplicated work which seems unnecessary.